### PR TITLE
DOC-2024: Document the new open source plugin, Accordion.

### PR DESCRIPTION
**Pull request by brian-forte**
Ticket: DOC-2024, Document the new open source plugin, Accordion.

Changes:
* Added required files:
	* `/modules/ROOT/partials/commands/accordion-cmds.adoc`.
	* `/modules/ROOT/partials/events/accordion-events.adoc`.
	* `/modules/ROOT/partials/menu-item-ids/accordion-menu-items.adoc`.
	* `/modules/ROOT/partials/toolbar-button-ids/accordion-toolbar-buttons.adoc`.
	* `/modules/ROOT/pages/accordion.adoc`.
	* `/modules/ROOT/partials/configuration/details_initial_state.adoc`.
	* `/modules/ROOT/partials/configuration/details_serialized_state.adoc`.
	* `/modules/ROOT/examples/live-demos/accordion/index.html`.
	* `/modules/ROOT/examples/live-demos/accordion/index.js`.
* Edited required files:
	* `/modules/ROOT/nav.adoc`.
	* `/modules/ROOT/pages/available-menu-items.adoc`.
	* `/modules/ROOT/pages/available-toolbar-buttons.adoc`.
* Added documentation of Accordion plugin to these files.

Pre-checks:
- [x] Branch prefixed with `feature/6/` or `hotfix/6/`
- [x] Changelog entry added
- [x] `modules/ROOT/nav.adoc` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed

### CODE REVIEW

File: modules/ROOT/examples/live-demos/accordion/index.html [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<textarea id="accordion">
+
+  {{logofordemoshtml}}
+
+  <h1 style="text-align: center;">The Accordions plugin</h1>
+
+  <p>This is a demo of the <strong>Accordion</strong> open source plugin.</p>
+
+  <details>
```
Comment (ShiridiGandham) [inline modules/ROOT/examples/live-demos/accordion/index.html:9]: Add this attribute to <details> tag -  class="mce-accordion"

- Change <details> to <details class="mce-accordion">

If class is not mentioned then user is unable to select the body content.

FYI @mkzhx

File: modules/ROOT/examples/live-demos/accordion/index.html [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    <summary>What does the <strong>Accordion</strong> plugin do?</summary>
+      <p>The Accordion plugin allows for the creation of collapsible sections in a document that can be expanded or collapsed to show or hide additional content.</p>
+  </details>
+
+  <details>
```
Comment (ShiridiGandham) [inline modules/ROOT/examples/live-demos/accordion/index.html:14]: Add this attribute to `<details>` tag -  class="mce-accordion"

- Change `<details>` to `<details class="mce-accordion">`

If class is not mentioned then user is unable to select the body content.

FYI @mkzhx

File: modules/ROOT/examples/live-demos/accordion/index.html [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    <summary>How does it present to an end-user?</summary>
+      <p>By default, when the <code>accordion</code> plugin is loaded, the command is available in the TinyMCE menu at <strong>Insert → Accordion</strong>.</p>
+      <p>Adding <code>accordion</code> to the <code>toolbar</code> in the TinyMCE configuration adds an Accordion icon to the TinyMCE toolbar.</p>
+      <p>And, when a specific collapsible section is selected, a pop-up menu presents offering two controls: one to collaps or expand the selected section; and one to delete the selected section.</p>
+  </details>
+
+  <details>
```
Comment (ShiridiGandham) [inline modules/ROOT/examples/live-demos/accordion/index.html:21]: Add this attribute to `<details>` tag -  class="mce-accordion"

- Change `<details>` to `<details class="mce-accordion">`

If class is not mentioned then user is unable to select the body content.

FYI @mkzhx

File: modules/ROOT/examples/live-demos/accordion/index.html (hunk 1 above)
Comment (brian-forte) [reply modules/ROOT/examples/live-demos/accordion/index.html]: So changed.

File: modules/ROOT/examples/live-demos/accordion/index.html (hunk 2 above)
Comment (brian-forte) [reply modules/ROOT/examples/live-demos/accordion/index.html]: So changed.

File: modules/ROOT/examples/live-demos/accordion/index.html (hunk 3 above)
Comment (brian-forte) [reply modules/ROOT/examples/live-demos/accordion/index.html]: So changed.

File: modules/ROOT/partials/configuration/details_initial_state.adoc [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+});
+----
+
+=== Example: using `details_initial_state` to set pre-configured `<details>` content in new {productname} documents to present as open
+
+[source,js]
+----
+tinymce.init({
+  selector: 'textarea',  // change this value according to your HTML
+  newdocument_content: '<details><summary>Accordion summary...</summary><p>Accordion body...</p></details>',
```
Comment (spocke) [inline modules/ROOT/partials/configuration/details_initial_state.adoc:40]: I really think we need to add that `initial_content` option that can set the initial content of the editor. I can see how that would make these types of examples so much easier. :smile:

**Review by spocke — APPROVED** [review]

File: modules/ROOT/examples/live-demos/accordion/index.html [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    <li>We also offer enterprise grade support as part of <a href="{{pricingpage}}">TinyMCE premium plans</a>.</li>
+  </ul>
+
+  <h2>Found a bug?</h2>
+
+  <p>If you think you have found a bug please create an issue on the <a href="https://github.com/tinymce/tinymce/issues">GitHub repo</a> to report it to the developers.</p>
+
+  <h2>Finally…</h2>
+
+  <p>Check out <a href="http://plupload.com" target="_blank">Plupload</a>, an upload solution that features HTML5 upload support.</p>
```
[the remark this answers is not in the document]
Comment (brian-forte) [reply modules/ROOT/examples/live-demos/accordion/index.html:44]: That section is part of the boilerplate for creating new interactive examples. It’s, nonetheless, a question I’ve half-thought to raise.

File: modules/ROOT/examples/live-demos/accordion/index.html [hunk 6]
```diff
@@ -0,0 +1,49 @@
+<textarea id="accordion">
+
+  {{logofordemoshtml}}
+
+  <h1 style="text-align: center;">The Accordions plugin</h1>
```
Comment (kemister85) [inline modules/ROOT/examples/live-demos/accordion/index.html:5]:
```suggestion
  <h1 style="text-align: center;">The Accordion plugin</h1>
```

Comment (kemister85) [reply modules/ROOT/examples/live-demos/accordion/index.html:5]: @brian-forte Not sure if this is a typo?

File: modules/ROOT/pages/accordion.adoc [hunk 7]
```diff
@@ -0,0 +1,64 @@
+= Accordion plugin
+:navtitle: Accordion
+:description: Creates sections that can be expanded or collapsed to show or hide content.
+:description_short: Creates expandable and collapsible sections.
+:keywords: plugin, accordion, collapse, expand, show, reveal, hide.
+:pluginname: Accordion
+:plugincode: accordion
+:plugincategory: opensource
+
+include::partial$misc/admon-requires-6.5v.adoc[]
```
Comment (kemister85) [inline modules/ROOT/pages/accordion.adoc:10]: @brian-forte Do we need to update the admon-requires to reflect the version bump to `6.5.1`

File: modules/ROOT/partials/commands/accordion-cmds.adoc [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+tinymce.activeEditor.execCommand('InsertAccordion');
+
+// sets the accordion at the current position to open
+tinymce.activeEditor.execCommand('ToggleAccordion', false, true);
+
+// sets all accordions in the current document to closed
+tinymce.activeEditor.execCommand('ToggleAllAccordions', false, false);
+
+// removes the accordion at the current position
+tinymce.activeEditor.execCommand('RemoveAccordion');
```
Comment (kemister85) [inline modules/ROOT/partials/commands/accordion-cmds.adoc:23]: @brian-forte Are we making these `execCommands` visible?

File: modules/ROOT/partials/configuration/details_initial_state.adoc [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  details_initial_state: 'expanded',
+});
+----
+
+NOTE
+//.Accordion plugin not required
```
Comment (kemister85) [inline modules/ROOT/partials/configuration/details_initial_state.adoc:46]: @brian-forte Should the `//` be removed?

File: modules/ROOT/partials/configuration/details_serialized_state.adoc [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+  selector: 'textarea',  // change this value according to your HTML
+  plugins: 'preview', // added to this example for UX demonstration purposes; see the NOTE below
+  newdocument_content: '<details><summary>Accordion summary...</summary><p>Accordion body...</p></details>',
+  details_initial_state: 'expanded',
+  details_serialized_state: 'collapsed',
+});
+----
+
+NOTE
+//.Accordion plugin not required
```
Comment (kemister85) [inline modules/ROOT/partials/configuration/details_serialized_state.adoc:44]: @brian-forte Should the `//` be removed here also?

**Review by kemister85 — APPROVED** [review]

**Review by ArvinJ-H — APPROVED** [review]

File: modules/ROOT/partials/configuration/details_initial_state.adoc (hunk 9 above)
Comment (brian-forte) [inline modules/ROOT/partials/configuration/details_initial_state.adoc:46]:
```suggestion
```
Remove commented out title that did not render.

File: modules/ROOT/partials/configuration/details_serialized_state.adoc (hunk 10 above)
Comment (brian-forte) [inline modules/ROOT/partials/configuration/details_serialized_state.adoc:44]:
```suggestion
```
Remove commented out title that did not render.

File: modules/ROOT/partials/plugin-apis/accordion-apis.adoc [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+// what this example does
+tinymce.activeEditor.plugins.accordion.<api1>(<arguments1>);
+
+// what this example does
+tinymce.activeEditor.plugins.accordion.<api2>(<arguments2>);
+
+// what this example does
+tinymce.activeEditor.plugins.accordion.<api3>(<arguments3>);
+// what this example does
+tinymce.activeEditor.plugins.accordion.<api4>(<arguments4>);
```
[the remark this answers is not in the document]
Comment (brian-forte) [reply modules/ROOT/partials/plugin-apis/accordion-apis.adoc:23]: No. The API-specific files are placeholder.

As it stands, there is no Accordion API, but the plugin will likely get an API in a future update.

**Pull request merged**